### ISSUE TITLE
Splash screen nevers goes away if location permission denied in Expo bare workflow project

### ISSUE THREAD

**Issue by srmagura** [transistorsoft/react-native-background-geolocation#1373]
## Your Environment
* Plugin version: 4.1.1
* Platform: iOS only. Not an issue on Android.
* OS version: iOS 12.5.4
* Device manufacturer / model: iPhone 6
* React Native version (`react-native -v`): 0.63.4
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->
useEffect(() => {
    BackgroundGeolocation.ready(
      {
        // Geolocation Config
        desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
        distanceFilter: 10,
        // Activity Recognition
        stopTimeout: 1,
        // Application config
        debug: true, // <-- enable this hear sounds for background-geolocation life-cycle.
        logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
        stopOnTerminate: false, // <-- Allow the background-service to continue tracking when user closes the app.
        startOnBoot: true, // <-- Auto start tracking when device is powered-up.

        // HTTP / SQLite config
        autoSync: false, // <-- [Default: true] Set true to sync each location to server as it arrives.
      },
      state => {
        console.log(
          '- BackgroundGeolocation is configured and ready: ',
          state.enabled,
        );

        if (!state.enabled) {
          ////
          // 3. Start tracking!
          //
          BackgroundGeolocation.start(function () {
            console.log('- Start success');
          });
        }
      },
    );

    return () => {
      BackgroundGeolocation.removeListeners();
    };
  }, []);
```

## Expected Behavior
When I open my app, I expect the splash screen to go away once the app has loaded, regardless of my permission settings. This should work in both normal React Native projects (`react-native init`) and Expo bare workflow projects (`expo init` with the "minimal" bare workflow template).

## Actual Behavior
When I open my normal React Native project with `react-native-background-geolocation` started and location permission denied, I am prompted to grant the app location permission, and when I hit cancel, the splash screen goes away as desired.

When I open my Expo bare workflow project with `react-native-background-geolocation` started and location permission denied, I am prompted to grant the app location permission, but the splash screen never goes away even after I hit cancel. This is only an issue if location permission is denied. The app opens successfully if I grant the location permission. With location permission denied, I get the warning "No native splash screen registered for given view controller. Call 'SplashScreen.show' for given view controller first." in Metro.

I think this may be related to https://github.com/expo/expo/issues/7689. The [comment by marcusmc](https://github.com/expo/expo/issues/7689#issuecomment-899049672) is particularly interesting – it looks like showing a modal while the splash screen is visible causes issues. (`react-native-background-geolocation` is showing a permissions modal while the splash screen is open.)

## Steps to Reproduce
1. Clone the minimal repro: https://github.com/srmagura/BackgroundLocationRepro
2. Follow the instructions in the README to run the normal React Native project on your iPhone. Close the app, set location permission to "Never" in the Settings app, and reopen the app. When prompted to grant location permission, press Cancel. The splash screen goes away as desired.
3. Follow the instructions in the README to run the Expo project on your iPhone. Close the app, set location permission to "Never" in the Settings app, and reopen the app. When prompted to grant location permission, press Cancel. The splash screen never goes away.

## Context
I am trying to utilize `react-native-background-geolocation` in an Expo bare workflow project while making my app functional for all users, even if they have denied the location permission.

I've already tried calling `BackgroundGeolocation.stop()` in the `onProviderChange` event handler if location permission was denied. This did not help.

## Debug logs
I clicked "View Device Logs" in Xcode for my iPhone but there were no logs related to either of my sample apps, since the app is not actually crashing. Let me know if there are additional logs that would be helpful to you.

**Comment by srmagura** [comment]
The more I think about this, the more it seems like the real issue is that Expo doesn't like it when a modal is displayed when the splash screen is up. I.e. the problem isn't really with `react-native-background-geolocation` at all.

I will work on a reproduction of the issue that doesn't involve `react-native-background-geolocation` and hopefully report it to Expo.

As a short term workaround, I will try to remove `expo-splash-screen` from my sample Expo project since that package seems to be the most likely source of the problem.

**Comment by srmagura** [comment]
I was able to reproduce this in both managed and bare Expo projects without `react-native-background-location` involved whatsoever, see https://github.com/expo/expo/issues/14099. I do believe this is an Expo issue, though a workaround could be implemented in `react-native-background-location`.

I'll leave this issue open for visibility but will not complain if you decide to close it. Thanks.

**Comment by srmagura** [comment]
This is fixed in `expo-splash-screen@0.11.4`!
